### ISSUE TITLE
Capabilities error in 1.4 module

### ISSUE THREAD

**Issue by Guite** [Guite/MostGenerator#691]
@rallek got the following error message when installing a module:

```
Capabilities-Eigenschaften müssen ein Array sein in der Form array(capability => array(version => string, ... => ... in RK\InfoModule\InfoModuleVersion
```

Probably caused by recent addition of `categorizable`.

The corresponding code looks like this:

```
        $meta['capabilities'] = array(
              'categorizable' => array(
                  'RK\InfoModule\Entity\informationEntity',
              ),
              HookUtil::SUBSCRIBER_CAPABLE => array('enabled' => true)
        );
```

@cmfcmf @craigh any idea?

**Comment by rallek** [comment]
I got this error while opening the extension module already

**Comment by craigh** [comment]
This is a generated 1.4 module? using what version of MOST? I thought there was no compatible release?

**Comment by rallek** [comment]
webgen. Just a try

**Comment by Guite** [comment]
@craigh the webgen allows always regenerating your model using the latest git version of the generator. You may read http://modulestudio.de/en/news/modulestudio-launches-service-for-web-based-generation.html for more details.

**Comment by craigh** [comment]
ok - please translate the error.

**Comment by Guite** [comment]
somehow Christian must have gotten the new Routes module, too ;-)

**Comment by rallek** [comment]
raw translation: 

Capabilities-properties must be an arry in the format(capability => array(version => string, ... => ... in RK\InfoModule\InfoModuleVersion

**Comment by cmfcmf** [comment]
```
$meta['capabilities'] = array(
              'categorizable' => array(
                  'blubb' => 'RK\InfoModule\Entity\informationEntity',
              ),
              HookUtil::SUBSCRIBER_CAPABLE => array('enabled' => true)
        );
```
should work.

**Comment by cmfcmf** [comment]
(replace "blubb" by something more.. intelligent.)

**Comment by craigh** [comment]
That isn't how I originally intended it. it should be
```json
      "capabilities": {
        "hook_subscriber": {"enabled": true},
        "searchable": {"class": "Zikula\\PagesModule\\Helper\\SearchHelper"},
        "categorizable": ["Zikula\\PagesModule\\Entity\\PageEntity"],
        "user": {"route": "zikulapagesmodule_user_index"},
        "admin": {"route": "zikulapagesmodule_admin_index"}
      },
````

so that there can be multiple entities in the array. e.g.
```json
        "categorizable": [
            "Zikula\\PagesModule\\Entity\\PageEntity",
            "Zikula\\PagesModule\\Entity\\FooEntity",
            "Zikula\\PagesModule\\Entity\\BarEntity"
        ],
```

if that isn't working then there is a problem somewhere.

**Comment by cmfcmf** [comment]
Capabilities must be Arrays of Arrays, this is how it is designed. Search
for "capabilities must be" in the core.

Craig Heydenburg <notifications@github.com> schrieb am Mo., 21. Sep. 2015
01:48:

**Comment by craigh** [comment]
> Capabilities must be Arrays of Arrays,

it IS. It just doesn't have associative keys in the final array. see below.

```php
$meta['capabilities'] = array(
              'categorizable' => array('RK\InfoModule\Entity\informationEntity'),
              HookUtil::SUBSCRIBER_CAPABLE => array('enabled' => true)
        );
```

**Comment by Guite** [comment]
So this would be correct?

```
$meta['capabilities'] = array(
              'categorizable' => array(
                  'information' => 'RK\InfoModule\Entity\InformationEntity',
                  'otherEntity' => 'RK\InfoModule\Entity\OtherEntityEntity'
              )
        );
```

**Comment by cmfcmf** [comment]
> it IS. It just doesn't have associative keys in the final array. see below.

Ok, it must have associative keys then. See https://github.com/zikula/core/blob/master/src/lib/legacy/Zikula/AbstractVersion.php#L505-505

**Comment by cmfcmf** [comment]
@guite Yes this should work.

**Comment by Guite** [comment]
Fixed in fa881e41c9cabbbcf0c9a635db9da3a3f851acc5
